modules/wallet/defrag.go

```diff
@@ -0,0 +1,64 @@
+package wallet
+
+import (
+	"sort"
+
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	// defragThreshold is the number of outputs a wallet is allowed before it is
+	// defragmented.
+	defragThreshold = 50
+
+	// defragBatchSize defines how many outputs are combined during one defrag.
+	defragBatchSize = 40
+)
+
+// defragWallet computes the sum of the 15 largest outputs in the wallet and
+// sends that sum to itself, effectively defragmenting the wallet. This defrag
+// operation is only performed if the wallet has greater than defragThreshold
+// outputs.
+func (w *Wallet) defragWallet() {
+	if len(w.siacoinOutputs) < defragThreshold {
+		return
+	}
+
+	// grab all the outputs from the wallet and sort them from
+	// largest -> smallest
+	var so sortedOutputs
+	for scoid, sco := range w.siacoinOutputs {
+		so.ids = append(so.ids, scoid)
+		so.outputs = append(so.outputs, sco)
+	}
+	sort.Sort(sort.Reverse(so))
+
+	// choose the defragBatchSizeth largest outputs from the wallet and sum them
+	defragOutputs := so.outputs[:defragBatchSize]
+	var totalOutputValue types.Currency
+	for _, output := range defragOutputs {
+		totalOutputValue = totalOutputValue.Add(output.Value)
+	}
+
+	w.log.Printf("performing wallet defragmentation transaction: %v outputs, %v total value\n", len(defragOutputs), totalOutputValue)
```

Comment:
But, really this log should be merged with the one below, no need to make two logging statements per defrag, just one is fine (one if success, a different one for each type of error (I think only 1 type possible right now))